plugins/gossip/parameters.go

```diff
@@ -11,6 +13,14 @@ type ParametersDefinition struct {
 
 	// MissingMessageRequestRelayProbability defines the probability of missing message requests being relayed to other neighbors.
 	MissingMessageRequestRelayProbability float64 `default:"0.01" usage:"the probability of missing message requests being relayed to other neighbors"`
+
+	MessagesRateLimit messagesLimitParameters
+}
+
+type messagesLimitParameters struct {
+	Interval        time.Duration `default:"1s" usage:"the time interval for which we count the messages rate"`
+	Limit           int           `default:"200" usage:"the default limit of messages per interval"`
+	DuringSyncLimit int           `default:"5000" usage:"the limit of messages per interval during node sync"`
```

Comment:
This is highly dependent on the bandwidth and power of nodes and I think very hard to specify meaningfully. How about instead of having another parameter simply counting down whenever we send a message request to a neighbor?

Reply:
You mean another rate counter for sent messages requests?

Reply:
No instead of having a separate limit while syncing I'd do the following per neighbor:
- count up if we receive a message
- count down if request a message (this should only happen during syncing/solidification) 

In that way we do not need to specify `DuringSyncLimit` a priori.

Reply:
My original idea was almost the same as yours (I just thought to increment the limit instead of count down on messages request)
It seems a little bit fragile to me and might rate limit some healthy peers during the sync. The questions I had:
1. When we count down/increment the limit we basically give "credit" to a peer. Why do we give it for the rate counter "interval" amount of time (1s) Why not less or more? What if takes more time for a peer to send the request message back to us.
2. With the rate counting algorithm you proposed (Fixed Window), we lose the credit for the peer when we reset the timer. Is it okay still?

Reply:
1. Good question, it's somewhat arbitrary I guess. We'd just need to find a good estimate on when the neighbor answers as to make sure that it does not cause rate limiting healthy peers. We can probably assume that a message request is answered within 1 second. RTT should be fairly fast between direct neighbors. 
2. Let's leave a sliding window approach for now.

So seeing the benefits of simpler concept, simplified code, and one less parameter to specify I'd just go for it (with 1 second for now). If we get false positives during syncing we can still change the policy or increase the limit for a longer period of time.

Reply:
Okay, let's try and see. This is indeed a more elegant solution.